src/Appium.Net/Appium/Service/AppiumCommandExecutor.cs

```diff
@@ -53,14 +54,16 @@ internal AppiumCommandExecutor(AppiumLocalService service, TimeSpan timeForTheSe
             ClientConfig = clientConfig;
         }
 
-        public Response Execute(Command commandToExecute)
+        public Response Execute(Command commandToExecute) => this.ExecuteAsync(commandToExecute).GetAwaiter().GetResult();
```

Comment:
Use `Task.Run(ExecuteAsync(commandToExecute)).GetAwaiter().GetResult()` to avoid deadlocks.